core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -78,12 +86,12 @@ trait SparkDataFrameConverter extends Logging with ConverterUtils {
   def perSQLPartition(types: Seq[(Seq[Int], StructField, Byte)])
                      (keyName: String, vecTypes: Array[Byte], uploadPlan: Option[immutable.Map[Int, NodeDesc]])
                      (context: TaskContext, it: Iterator[Row]): (Int, Long) = {
+    val asArr = it.toArray[Row] // need to buffer the iterator in order to get its length
```

Comment:
bad name; also, why do you want arrays all around? a list would be enough, I think
(and I wonder why Iterators are being passed all around; a Seq would be way better)